packages/datetime/src/dateInput.tsx

```diff
@@ -463,24 +473,33 @@ export class DateInput extends AbstractPureComponent2<DateInputProps, IDateInput
             const isChangeMonthButtonDisabled = isChangeMonthEvt && (eventTarget as HTMLButtonElement).disabled;
             const isDayPickerDayEvt = eventTarget.classList.contains(Classes.DATEPICKER_DAY);
             if (!isChangeMonthButtonDisabled && !isDayPickerDayEvt) {
+                if (eventTarget === this.firstTabbableElement) {
+                    this.inputElement?.focus();
+                    e.preventDefault();
+                }
                 this.handleClosePopover();
             }
         } else if (relatedTarget != null) {
-            this.unregisterPopoverBlurHandler();
+            this.unregisterPopoverBlurHandlers();
+            this.firstTabbableElement = this.getFirstTabbableElement();
+            this.firstTabbableElement?.addEventListener("blur", this.handlePopoverBlur);
```

Comment:
this doesn't seem exactly right... I feel like the popover should stay open after a SHIFT+TAB keypress on the first tabbable element?

for example, CLICK, TAB, SHIFT+TAB interaction here:

![2021-09-24 12 20 26](https://user-images.githubusercontent.com/723999/134708673-115ded5e-d0ce-425c-8695-0a460412f34c.gif)

Reply:
I could go either way, though... open to other ideas